pkg/sql/delegate/show_regions.go

```diff
@@ -13,11 +13,18 @@ package delegate
 import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltelemetry"
-	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
 )
 
 // delegateShowRanges implements the SHOW REGIONS statement.
 func (d *delegator) delegateShowRegions(n *tree.ShowRegions) (tree.Statement, error) {
 	sqltelemetry.IncrementShowCounter(sqltelemetry.Regions)
-	return nil, unimplemented.New("show regions", "implementation pending")
+	query := `SELECT 
```

Comment:
can you run this query through https://sqlfum.pt/ to make this easier to read?
do we want availability zones to be grouped as an array, i.e.

```
region       |       az
us-west-1 | {"us-west-1a", "us-west-1b"}
us-west-2 | {"us-west-1c", "us-west-1d"}

```

Reply:
I like the array idea, but I'm wondering if the brackets and quotes are necessary.  What I have now is this:

`    region    | availability_zones
--------------+---------------------
  asia-east   | b
  europe-east | a
  us-east     | a,b
  us-west     | a
(4 rows)
`
In most cases, the az will be a single character, which leads me to believe that the array formatting might be overkill, unless that's the convention elsewhere.

I'm inclined to skip the dbs column, and have customers use FOR DATABASE.  Reasonable?

Reply:
the brackets are formatted that way on certain systems -- as long as it's an array it's fine

> I'm inclined to skip the dbs column, and have customers use FOR DATABASE. Reasonable?

i guess my question is then -- how do i quickly query which databases a region contains? e.g. i want availability zone "a" to map to "database c,d".

SHOW REGIONS might not be the right place, but I think it's something we should consider.

Reply:
Added the array.

Agree about a place for the database.  I'm thinking that should be someplace different though.  Happy to discuss this further, or push this and revisit later.  Your call...

Reply:
as long as it's on the spec somewhere or an issue i don't mind!

thanks!